app/utils/schemaUtils.ts

```diff
@@ -141,3 +147,53 @@ export const createCbcSchemaData = (jsonData) => {
 
   return dataBySection;
 };
+
+type CommunitySourceData = {
+  readonly bcGeographicName: string;
+  readonly geographicNameId: number;
+  readonly regionalDistrict: string | null;
+  readonly economicRegion: string | null;
+};
+
+export const generateGeographicNamesByRegionalDistrict = (
+  allCommunitiesSourceData: readonly CommunitySourceData[]
+) => {
+  const regionalDistrictGeographicNamesDict = {};
+  allCommunitiesSourceData.forEach((community) => {
+    const { regionalDistrict, bcGeographicName, geographicNameId } = community;
+    if (!regionalDistrictGeographicNamesDict[regionalDistrict]) {
+      regionalDistrictGeographicNamesDict[regionalDistrict] = new Set();
+    }
+    regionalDistrictGeographicNamesDict[regionalDistrict].add({
+      label: bcGeographicName,
+      value: geographicNameId,
+    });
+  });
+  return regionalDistrictGeographicNamesDict;
+};
+
+export const generateRegionalDistrictsByEconomicRegion = (
+  allCommunitiesSourceData: readonly CommunitySourceData[]
+) => {
+  const economicRegionRegionalDistrictsDict = {};
+  allCommunitiesSourceData.forEach((community) => {
+    const { economicRegion, regionalDistrict } = community;
+    if (!economicRegionRegionalDistrictsDict[economicRegion]) {
+      economicRegionRegionalDistrictsDict[economicRegion] = new Set();
+    }
+    economicRegionRegionalDistrictsDict[economicRegion].add(regionalDistrict);
+  });
+
+  return economicRegionRegionalDistrictsDict;
+};
+
+export const getAllEconomicRegionNames = (
```

Comment:
Noticed while populating `Geographic names` the geographic names are populated based on RDs not RDs and ERs
For example: 
when using ER --> `Banff--Jasper--Rocky Mountain House`
![image](https://github.com/user-attachments/assets/e83c188b-4551-4f06-a15d-890aabb47233)
but should only have `Crowsnest`
And when added it gets added errorneously under `Kootney` not `Banff--Jasper--Rocky Mountain House`
![image](https://github.com/user-attachments/assets/451d8969-99c4-425c-b416-3b26ce27c8a1)

Reply:
I've added an update for this, I am curious about what you mean by it being added to Kootenay though - let me know if this behaviour persists with the updates I made

Reply:
Sure let me check, what I meant was even if I select `Banff--Jasper--Rocky Mountain House` as the ER, since Baker is not listed as part of it, it gets added to ER -> `Kootney`